### DISCUSSION TITLE
External spellchecker

### DISCUSSION THREAD

**Discussion by HectorB-2020** [ONLYOFFICE/DocumentServer#1882] [Q&A]
Hello there!
I'm wondering if it would be technically possible to integrate any external commercial spellchecker into OnlyOffice products. I clearly recall and cherish times when "Orfo" was a part of Microsoft Office. So I've been dreaming of making it a part of Only Office.
Did you make any attempts to negotiate commercial cooperation with the company which produces that application? At least for commercial OnlyOffice.

**Comment by ShockwaveNN** [comment]
Hi, I think it's possible to be done as a plugin if the spellchecker service provides some kind of API

But I doubt that it will be a part of default distribution since it's probably will be not open source

**Reply by HectorB-2020** [reply]
I certainly don't expect it to be an open source tool. At least "Orfo" has never been an Open Source.
So having said that, does OnlyOffice have an API or Plugin or which API should the other solution export to become plug-able into OnlyOffice?

**Reply by ShockwaveNN** [reply]
You can find out all the information about plugin API [here](https://api.onlyoffice.com/plugin/basic)

And you can take a look at [`glavred`](https://github.com/ONLYOFFICE/plugin-glavred) plugin which provide basic syntax and orthographic check

But I think it's really depend on service and which API this Orfo (never heard about it) provides
